etc/travis/DeploymentTester.py

```diff
@@ -0,0 +1,115 @@
+import pycurl
+import sys
+import time
+from pycurl import Curl
+
+from azure.mgmt.resource import ResourceManagementClient
+from msrestazure.azure_active_directory import ServicePrincipalCredentials
+
+from travis.Configuration import Configuration
+
+
+class DeploymentTester:
+    @staticmethod
+    def elapsed(since):
+        elapsed = int(time.time() - since)
+        elapsed = '{:02d}:{:02d}:{:02d}'.format(elapsed // 3600, (elapsed % 3600 // 60), elapsed % 60)
+        return elapsed
+
+    def __init__(self):
+        self.config = Configuration()
+        self.deployment = None
+
+        self.credentials = None
+        """:type : ServicePrincipalCredentials"""
+
+        self.resource_client = None
+        """:type : ResourceManagementClient"""
+
+    def run(self):
+        self.check_configuration()
+        self.login()
+        self.create_resource_group()
+        self.validate()
+        self.deploy()
+        self.moodle_smoke_test()
+        print('\n\nJob done!')
+
+    def check_configuration(self):
+        print('\nChecking configuration...')
+        if not self.config.is_valid():
+            print('No Azure deployment info given, skipping test deployment and exiting.')
+            print('Further information: https://github.com/Azure/Moodle#automated-testing-travis-ci')
+            sys.exit()
+        print("(all check)")
+
+    def login(self):
+        print('\nLogging in...')
+        self.credentials = ServicePrincipalCredentials(
+            client_id=self.config.client_id,
+            secret=self.config.secret,
+            tenant=self.config.tenant_id,
+        )
+        self.resource_client = ResourceManagementClient(self.credentials, self.config.subscription_id)
```

Comment:
So this seems to be the only place where the subscription ID is needed here, just because of this Python SDK API's requirement. However, a service principal is always associated with an Azure resource ID, which always includes a subscription, so that's why we didn't need to specify a subscription ID in the bash script. I wonder if we can somehow avoid having to provide a subscription ID here? Not a big deal, but just wondering...

Reply:
I am not very familiar with the the Azure API in Python, but I'd guess it is possible to find out somehow. I'll investigate.

Reply:
Just mentioning again that this is not a must. If it takes more than half an hour of your time, please disregard this comment.